src/package/pywinrt/projection/CMakeLists.txt

```diff
@@ -1,32 +1,73 @@
 cmake_minimum_required(VERSION 3.12)
 
-if(NOT DEFINED PACKAGE_ROOT_PATH)
-    message(FATAL_ERROR "PACKAGE_ROOT_PATH not defined")
-endif()
+function(ADD_OBJECT_DEPENDS file dependency)
+    get_source_file_property(cur_obj_depends ${file} OBJECT_DEPENDS)
+    if (${cur_obj_depends} STREQUAL "NOTFOUND")
+        set(new_obj_depends "${pch_output_file_name}")
+    else()
+        set(new_obj_depends "${cur_obj_depends};${dependency}")
+    endif()
+    set_source_files_properties(${file}  PROPERTIES OBJECT_DEPENDS ${new_obj_depends})
+endfunction(ADD_OBJECT_DEPENDS)
+
+function(TARGET_CONFIG_MSVC_PCH target pch_cpp pch_header)
+
+    get_target_property(target_name ${target} NAME)
+    set(pch_output_file_name "${target_name}.pch")
+    set(pch_output_file "\"${CMAKE_CURRENT_BINARY_DIR}/${pch_output_file_name}\"")
+
+    get_target_property(sources ${target} SOURCES)
+    foreach(file ${sources})
+        get_filename_component(filename ${file} NAME)
+        if (${filename} STREQUAL ${pch_cpp})
+            set_source_files_properties(${file}
+                PROPERTIES
+                COMPILE_FLAGS " /Yc${pch_header} /Fp${pch_output_file}"
+                OBJECT_OUTPUTS ${pch_output_file_name})
+        else()
+            set_source_files_properties(${file} PROPERTIES COMPILE_FLAGS " /Yu${pch_header} /Fp${pch_output_file}")
+            ADD_OBJECT_DEPENDS(${file} ${pch_output_file_name})
+        endif()
+    endforeach()
+
+    set_target_properties(${target} PROPERTIES PCH_OUTPUT_FILE ${pch_output_file})
+    set_property(GLOBAL APPEND PROPERTY PCH_OUTPUT_FILES ${pch_output_file})
+
+endfunction(TARGET_CONFIG_MSVC_PCH)
+
+file(GLOB sources "${CMAKE_CURRENT_SOURCE_DIR}/pywinrt/winrt/src/*.cpp")
+foreach(file ${sources})
+    message(${file})
+endforeach()
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-add_compile_options(/await /bigobj /GL /GR- /permissive-)
+
+# TODO: re-enable /GL and /LTCG when we get a bigger build machine
+add_compile_options(/await /bigobj /GR- /permissive-)
```

Comment:
this applies to both x86 and x64 builds?  workaround just means slightly longer build times?

Reply:
Yes, this applies to both x86 and x64. I was running out of heap space compiling the full projection for both archtiectures w/ LTCG turned on. I am looking into getting access to a build agent with more memory for internal builds, but I am not going to hold up this PR for that.